src/libexpr/include/nix/expr/parser-state.hh

```diff
@@ -253,11 +254,17 @@ ParserState::stripIndentation(const PosIdx pos, std::vector<std::pair<PosIdx, st
     bool atStartOfLine = true; /* = seen only whitespace in the current line */
     size_t minIndent = 1000000;
     size_t curIndent = 0;
-    for (auto & [i_pos, i] : es) {
+    std::vector<int> nrIndentedLines(es.size(), 0);
+    std::vector<bool> perfectPreallocate(es.size(), true);
```

Comment:
Aren't these allocations killing most of the benefits of the buffer allocator?

Reply:
I think at this point the main goal is trying not to leak as much memory for exprs. The performance benefits are supposed to come later.

Reply:
Indeed these are short-lived allocations only for this function so they're not going to kill any long-term benefits of not over-allocating.